bird_cloud_gnn/cross_validation.py

```diff
@@ -0,0 +1,65 @@
+"""Helper functions for cross validation.
+"""
+import numpy as np
+from dgl.dataloading import GraphDataLoader
+from sklearn.model_selection import StratifiedKFold
+from torch.utils.data.sampler import SubsetRandomSampler
+from tqdm import tqdm
+from bird_cloud_gnn.gnn_model import GCN
+
+
+# pylint: disable=too-many-arguments, too-many-locals
+def kfold_evaluate(
+    dataset,
+    h_feats=16,
+    n_splits=5,
+    learning_rate=0.01,
+    num_epochs=100,
+    batch_size=512,
+):
+    """
+    Evaluate the model on a dataset using StratifiedKFold.
+
+    Args:
+        dataset (RadarDataset): The dataset
+        h_feats (int, optional): The number of hidden features of the model
+        n_splits (int, optional): Number of folds. Defaults to 5.
+        learning_rate (float, optional): Learning rate. Defaults to 0.01.
+        num_epochs (int, optional): Training epochs. Defaults to 20.
+        batch_size (int, optional): Batch size used in the data loaders. Defaults to 512.
+    """
+
+    labels = np.array(dataset.labels)
+    # Initialize a stratified k-fold splitter
+    kfold = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=42)
```

Comment:
Is it desirable to fix the stae for the random number generator here, I think we don't do that for radarDataSet it would be good to be consistent

Reply:
I created an issue to deal with that separately.